hangman/lib/hangman.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Hangman do
+  @moduledoc """
+  Documentation for Hangman.
+  Christian Ventura
+  """
+
+  #the defined struct
+  defstruct game_state: :initializing, turns_left: 7, letters: [], used: [], last_guess: "", word: ""
+
+
+  def word_to_blank([_ | []]) do
+    ["_"]                                #reached the end of the string, so return
+  end
+ 
+  def word_to_blank([_ | tail]) do
+    ["_" | word_to_blank(tail)]   # turn a string into underscores
+  end
+
+  #replace underscores with previous guesses. For the last character in the string
+  def word_to_blank([guess_head | []], [actual_head | []], guess) do
+    if ( guess_head == actual_head) do
+      [guess_head]
+    else 
+      if (actual_head == guess) do
+        [guess]
+      else
+        ["_"]
+      end
+    end
+  end
+
+  #Replace underscores with correct guesses. Remember to keep previous guesses
+  def word_to_blank([guess_head | guess_tail], [actual_head | actual_tail], guess) do
+    if ( guess_head == actual_head) do 
+      [guess_head | word_to_blank(guess_tail, actual_tail, guess)]
+    else 
+      if (actual_head == guess) do
+        [guess | word_to_blank(guess_tail, actual_tail, guess)]
+      else
+        ["_" | word_to_blank(guess_tail, actual_tail, guess)]
+      end
+    end
+  end
+
+  def new_game() do
+    word = String.graphemes(Dictionary.random_word())   #turn string into characters
```

Comment:
This doesn't run" line 46 sets `word` to a list of graphemes, and then line 47 seems to do a string replace on it.